src/include/mallocMC/creationPolicies/Scatter_impl.hpp

```diff
@@ -132,6 +132,8 @@ namespace ScatterKernelDetail{
       //static const uint32 minChunkSize0 = pagesize/(32*32);
       static const uint32 minChunkSize1 = 0x10;
       static const uint32 HierarchyThreshold =  (pagesize - 2*sizeof(uint32))/33;
+      static const uint32 tmp_maxOPM = minChunkSize1 > HierarchyThreshold ? 0 : pagesize/(32*minChunkSize1 + 1);
```

Comment:
wow, good catch :+1: . I guess it was really late that night :sleeping: :crescent_moon: 
I think there are multiple errors in that line:
- for each full segment of the page, it takes `32*chunksize + sizeof(uint32)` byte. so the `+1` is flawed, since we count the number of bytes here:

`pagesize/(32*minChunkSize1 + sizeof(uint32))`
- for the (possible) **n** additional chunks, we need `n*chunksize+sizeof(32)`  byte, so there needs to be a round-up as well. It becomes:

`(pagesize + (32*minChunkSize1 + sizeof(uint32)) -1)/(32*minChunkSize1 + sizeof(uint32))`

Example1:
pagesize = 4096
minChunkSize = 16
-> space for 7 full segments (and their 7 onpagemasks), and 30 additional chunks (with 1 onpagemask)
`7*(32*16 + 4) + (30*16 + 4) = 4096`
The new formula gives **8** as a result, which is the necessary number of onpagemasks.